### PR TITLE
fix client-id naming to be unique for multiple consumers

### LINKED ISSUE #166: Client ID should not be reused across multiple Kafka consumer instances
- Version: v6.2.4/master
- Operating System: All
- Config File (if you have sensitive info, please remove it):
```
input {
kafka
{ topics => 'logstash_logs02' client_id => "oneapp.app.log.msp" group_id => "group02" consumer_threads => 20 }
}
```
- Sample Data: -
- Steps to Reproduce:
Run a Kafka input with multiple consumer threads.
You will see errors in the log like this:
```
M[?:com.ibm.streamsx.messaging.kafka.KafkaConsumerV9.produceTuples:-1] - Error processing messages: Error registering mbean kafka.consumer:type=consumer-node-metrics,client-id=consumer-1,node-id=node--1
org.apache.kafka.common.KafkaException: Error registering mbean kafka.consumer:type=consumer-node-metrics,client-id=consumer-1,node-id=node--1
at org.apache.kafka.common.metrics.JmxReporter.reregister(JmxReporter.java:159)
at org.apache.kafka.common.metrics.JmxReporter.metricChange(JmxReporter.java:77)
at org.apache.kafka.common.metrics.Metrics.registerMetric(Metrics.java:288)
at org.apache.kafka.common.metrics.Sensor.add(Sensor.java:177)
at org.apache.kafka.common.metrics.Sensor.add(Sensor.java:162)
at org.apache.kafka.common.network.Selector$SelectorMetrics.maybeRegisterConnectionMetrics(Selector.java:641)
at org.apache.kafka.common.network.Selector.poll(Selector.java:268)
at org.apache.kafka.clients.NetworkClient.poll(NetworkClient.java:270)
at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.clientPoll(ConsumerNetworkClient.java:303)
at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:197)
at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:187)
at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.awaitMetadataUpdate(ConsumerNetworkClient.java:126)
at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.ensureCoordinatorKnown(AbstractCoordinator.java:186)
at org.apache.kafka.clients.consumer.KafkaConsumer.pollOnce(KafkaConsumer.java:857)
at org.apache.kafka.clients.consumer.KafkaConsumer.poll(KafkaConsumer.java:829)
at com.ibm.streamsx.messaging.kafka.KafkaConsumerV9.produceTuples(KafkaConsumerV9.java:129)
at com.ibm.streamsx.messaging.kafka.KafkaConsumerV9$1.run(KafkaConsumerV9.java:70)
at java.lang.Thread.run(Thread.java:785)
at com.ibm.streams.operator.internal.runtime.OperatorThreadFactory$2.run(OperatorThreadFactory.java:137)
Caused by: javax.management.InstanceAlreadyExistsException: kafka.consumer:type=consumer-node-metrics,client-id=consumer-1,node-id=node--1
at com.sun.jmx.mbeanserver.Repository.addMBean(Repository.java:449)
at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.registerWithRepository(DefaultMBeanServerInterceptor.java:1910)
at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.registerDynamicMBean(DefaultMBeanServerInterceptor.java:978)
at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.registerObject(DefaultMBeanServerInterceptor.java:912)
at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.registerMBean(DefaultMBeanServerInterceptor.java:336)
at com.sun.jmx.mbeanserver.JmxMBeanServer.registerMBean(JmxMBeanServer.java:534)
at org.apache.kafka.common.metrics.JmxReporter.reregister(JmxReporter.java:157)
... 18 more
```
Try looking at JMX metrics for the consumers and you will only see one instance despite there being multiple consumers.
See https://github.com/logstash-plugins/logstash-input-kafka/blob/master/lib/logstash/inputs/kafka.rb#L269-L278 for the issue -- client IDs shouldn't be reused within the same process because they generate identically named metrics.

### PR DESCRIPTION
**Pull request by talevy**
Closes #166.

will port to v5.x

### CODE REVIEW

File: lib/logstash/inputs/kafka.rb [hunk 1]
```diff
@@ -230,6 +230,11 @@ def stop
     @runner_consumers.each { |c| c.wakeup }
   end
 
+  public
+  def inner_consumers
```
Comment (suyograo) [inline lib/logstash/inputs/kafka.rb]: Can we rename this to `kafka_consumers`

**Comment by talevy** [comment]
backported to 5.x here: https://github.com/logstash-plugins/logstash-input-kafka/pull/172

**Comment by seanalbert** [comment]
Could you backport this to 4.x please ?

**Comment by rodrigongarcia** [comment]
Could you backport this to 4.x please ?

**Pull request merged**